leetcode/robot-return-to-origin.js

```diff
@@ -0,0 +1,37 @@
+// https://leetcode.com/problems/robot-return-to-origin/
+
+/**
+ * @param {string} moves
+ * @return {boolean}
+ */
+
+var judgeCircle = function(moves) {
+    const movesCount = {
+        "U": 0,
+        "D": 0,
+        "L": 0,
+        "R": 0
+    }
+    
+    moves.split("").forEach(move => { //"U"
+        
+        move = move.toUpperCase(); 
+        
+        console.log(move);
+
+        // return movesCount[move] !== undefined ? movesCount[move] += 1 : false;
+        return !movesCount[move] ? false : movesCount[move] += 1;
+    })
+
+    console.log("moves count obj: ", movesCount)
+    
+    return movesCount["U"] === movesCount["D"] && movesCount["L"] === movesCount["R"] 
+};
+
+console.log(judgeCircle("ud")) // true
+// console.log(judgeCircle("UD")) // true
+// console.log(judgeCircle("UDD")) // false
+// console.log(judgeCircle("LL")) // false
```

Comment:
what about something that uses UDLR (all the moves?) code wise it's the same as testing UD and LR separately, but from a problem standpoint it's probably something we want to take a look at